subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -367,9 +367,36 @@ int fota_download_init(fota_download_callback_t client_callback)
 		return err;
 	}
 
+	first_fragment = true;
 	return 0;
 }
 
+int fota_download_cancel(void)
+{
+	int err = 0;
+
+	if (dlc.fd == -1) {
+		/* Download not started, aborted or completed */
+		err = -EALREADY;
```

Comment:
Just print the warning and return early with -EAGAIN, to aviod the extra indentation.

Reply:
Early return added